tensorflow_encrypted/session.py

```diff
@@ -0,0 +1,135 @@
+from typing import Dict, List, Optional, Any, Union
+import os
+from collections import Iterable, defaultdict
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.client import timeline
+
+__TFE_STATS__ = bool(os.getenv('TFE_STATS', False))
+__TFE_TRACE__ = bool(os.getenv('TFE_TRACE', False))
+__TFE_DEBUG__ = bool(os.getenv('TFE_DEBUG', False))
+__TENSORBOARD_DIR__ = str(os.getenv('TFE_STATS_DIR', '/tmp/tensorboard'))
+
+_run_counter = defaultdict(int)  # type: Any
+
+
+class TFESession():
+    """
+    Wrap a Tensorflow Session
+    """
+
+    def __init__(self, sess: tf.Session) -> None:
+        self.sess = sess
+
+    def __enter__(self) -> 'TFESession':
+        self.sess = self.sess.__enter__()
+        return self
+
+    def __exit__(self, exec_type, exec_value, exec_tb):
+        self.sess.__exit__(exec_type, exec_value, exec_tb)
+
+    def sanitize_fetches(self, fetches: Any) -> Union[List[Any], tf.Tensor, tf.Operation]:
```

Comment:
maybe rewrite this as a recursive function, think that would clean it up a lot and allow for deeper nesting

Reply:
Done